File: examples/java/BrebExample.java
package app.bloque.examples;

import app.bloque.sdk.BloqueSDK;
import app.bloque.sdk.UserSession;
import app.bloque.sdk.accounts.ActivateBrebKeyParams;
import app.bloque.sdk.accounts.BrebKeyAccount;
import app.bloque.sdk.accounts.BrebKeyType;
import app.bloque.sdk.accounts.BrebOperationResult;
import app.bloque.sdk.accounts.BrebResolvedKey;
import app.bloque.sdk.accounts.CreateBrebKeyParams;
import app.bloque.sdk.accounts.DeleteBrebKeyParams;
import app.bloque.sdk.accounts.DeleteBrebKeyResult;
import app.bloque.sdk.accounts.ResolveBrebKeyParams;
import app.bloque.sdk.accounts.SuspendBrebKeyParams;
import app.bloque.sdk.accounts.SuspendBrebKeyResult;
import app.bloque.sdk.accounts.ActivateBrebKeyResult;
import app.bloque.sdk.core.Mode;

import java.util.HashMap;
import java.util.Map;

/**
 * Java example: BRE-B key operations
 *
 * This example demonstrates how to create, resolve, suspend, activate,
 * and delete BRE-B keys using the Bloque SDK in Java.
 */
public class BrebExample {

    public static void main(String[] args) {
        BloqueSDK bloque = BloqueSDK.builder()
            .origin("{your-origin-here}")
            .apiKey("{your-api-key-here}")
            .mode(Mode.SANDBOX)
            .build();

        UserSession session = bloque.connect("nestor");

        System.out.println("=== Example 1: Create BRE-B Key ===");

        Map<String, Object> metadata = new HashMap<>();
        metadata.put("source", "example");
        metadata.put("channel", "sdk-java");

        BrebOperationResult<BrebKeyAccount> created = session.getAccounts()
            .getBreb()
            .createKey(new CreateBrebKeyParams(
                BrebKeyType.PHONE,
                "3123185778",
                "Pepito Silva",
                "ledger-account-breb-001",
                null,
                metadata
            ));

        System.out.println("Create error: " +
            (created.getError() != null ? created.getError().getMessage() : "null"));
        System.out.println("Account URN: " +
            (created.getData() != null ? created.getData().getUrn() : "null"));
        System.out.println("Remote Key ID: " +
            (created.getData() != null ? created.getData().getRemoteKeyId() : "null"));
        System.out.println("Status: " +
            (created.getData() != null ? created.getData().getStatus() : "null"));

        System.out.println("\n=== Example 2: Resolve BRE-B Key ===");

        BrebOperationResult<BrebResolvedKey> resolved = session.getAccounts()
            .getBreb()
            .resolveKey(new ResolveBrebKeyParams(BrebKeyType.PHONE, "3123185778"));

        System.out.println("Resolve error: " +
            (resolved.getError() != null ? resolved.getError().getMessage() : "null"));
        System.out.println("Resolution ID: " +
            (resolved.getData() != null ? resolved.getData().getResolutionId() : "null"));
        System.out.println("Owner name: " +
            (resolved.getData() != null && resolved.getData().getOwner() != null
                ? resolved.getData().getOwner().getName()
                : "null"));
        System.out.println("Receptor node: " +
            (resolved.getData() != null ? resolved.getData().getReceptorNode() : "null"));

        if (created.getData() != null) {
            String accountUrn = created.getData().getUrn();

            System.out.println("\n=== Example 3: Suspend BRE-B Key ===");

            BrebOperationResult<SuspendBrebKeyResult> suspended = session.getAccounts()
                .getBreb()
                .suspendKey(new SuspendBrebKeyParams(accountUrn));

            System.out.println("Suspend error: " +
                (suspended.getError() != null ? suspended.getError().getMessage() : "null"));
            System.out.println("Suspend status: " +
                (suspended.getData() != null ? suspended.getData().getStatus() : "null"));
            System.out.println("Upstream key status: " +
                (suspended.getData() != null ? suspended.getData().getKeyStatus() : "null"));

            System.out.println("\n=== Example 4: Activate BRE-B Key ===");

            BrebOperationResult<ActivateBrebKeyResult> activated = session.getAccounts()
                .getBreb()
                .activateKey(new ActivateBrebKeyParams(accountUrn));

            System.out.println("Activate error: " +
                (activated.getError() != null ? activated.getError().getMessage() : "null"));
            System.out.println("Activate status: " +
                (activated.getData() != null ? activated.getData().getStatus() : "null"));
            System.out.println("Upstream key status: " +
                (activated.getData() != null ? activated.getData().getKeyStatus() : "null"));

            System.out.println("\n=== Example 5: Delete BRE-B Key ===");

            BrebOperationResult<DeleteBrebKeyResult> deleted = session.getAccounts()
                .getBreb()
                .deleteKey(new DeleteBrebKeyParams(accountUrn));

            System.out.println("Delete error: " +
                (deleted.getError() != null ? deleted.getError().getMessage() : "null"));
            System.out.println("Deleted: " +
                (deleted.getData() != null ? deleted.getData().getDeleted() : "null"));
            System.out.println("Delete status: " +
                (deleted.getData() != null ? deleted.getData().getStatus() : "null"));
        }

        System.out.println("\n=== All BRE-B Examples Completed ===");
    }
}

File: examples/kotlin/BrebExample.kt
package app.bloque.examples

import app.bloque.sdk.BloqueSDK
import app.bloque.sdk.accounts.ActivateBrebKeyParams
import app.bloque.sdk.accounts.BrebKeyType
import app.bloque.sdk.accounts.CreateBrebKeyParams
import app.bloque.sdk.accounts.DeleteBrebKeyParams
import app.bloque.sdk.accounts.ResolveBrebKeyParams
import app.bloque.sdk.accounts.SuspendBrebKeyParams
import app.bloque.sdk.core.Mode
import app.bloque.sdk.identity.IndividualRegisterParams
import app.bloque.sdk.identity.UserProfile

/**
 * Kotlin example: BRE-B key operations
 *
 * This example demonstrates how to create, resolve, suspend, activate,
 * and delete BRE-B keys using the Bloque SDK in idiomatic Kotlin.
 */
fun main() {
    val bloque = BloqueSDK.create(
        origin = "{your-origin-here}",
        apiKey = "{your-api-key-here}",
        mode = Mode.SANDBOX
    )

    val session = bloque.register(
        "example-user-breb",
        IndividualRegisterParams(
            UserProfile(
                firstName = "Example",
                lastName = "User",
                email = "example-breb@example.com",
                phone = "+573123185778"
            )
        )
    )

    println("=== Example 1: Create BRE-B Key ===")

    val created = session.accounts.breb.createKey(
        CreateBrebKeyParams(
            keyType = BrebKeyType.PHONE,
            key = "3123185778",
            displayName = "Pepito Silva",
            ledgerId = "ledger-account-breb-001",
            metadata = mapOf(
                "source" to "example",
                "channel" to "sdk-kotlin"
            )
        )
    )

    println("Create error: ${created.error?.message}")
    println("Account URN: ${created.data?.urn}")
    println("Remote Key ID: ${created.data?.remoteKeyId}")
    println("Status: ${created.data?.status}")

    println("\n=== Example 2: Resolve BRE-B Key ===")

    val resolved = session.accounts.breb.resolveKey(
        ResolveBrebKeyParams(
            keyType = BrebKeyType.PHONE,
            key = "3123185778"
        )
    )

    println("Resolve error: ${resolved.error?.message}")
    println("Resolution ID: ${resolved.data?.resolutionId}")
    println("Owner name: ${resolved.data?.owner?.name}")
    println("Receptor node: ${resolved.data?.receptorNode}")

    created.data?.let { account ->
        println("\n=== Example 3: Suspend BRE-B Key ===")

        val suspended = session.accounts.breb.suspendKey(
            SuspendBrebKeyParams(accountUrn = account.urn)
        )

        println("Suspend error: ${suspended.error?.message}")
        println("Suspend status: ${suspended.data?.status}")
        println("Upstream key status: ${suspended.data?.keyStatus}")

        println("\n=== Example 4: Activate BRE-B Key ===")

        val activated = session.accounts.breb.activateKey(
            ActivateBrebKeyParams(accountUrn = account.urn)
        )

        println("Activate error: ${activated.error?.message}")
        println("Activate status: ${activated.data?.status}")
        println("Upstream key status: ${activated.data?.keyStatus}")

        println("\n=== Example 5: Delete BRE-B Key ===")

        val deleted = session.accounts.breb.deleteKey(
            DeleteBrebKeyParams(accountUrn = account.urn)
        )

        println("Delete error: ${deleted.error?.message}")
        println("Deleted: ${deleted.data?.deleted}")
        println("Delete status: ${deleted.data?.status}")
    }

    println("\n=== All BRE-B Examples Completed ===")
}
